org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/HandleCryptosystemTextbook.java

```diff
@@ -0,0 +1,974 @@
+			maxBytesPerBlock = byteLenOfPlaintext;
+		
+		
+		int iblocklengthFinal = guiHandler.blocklength;
+		if((byteLenOfPlaintext * 2) < iblocklengthFinal)
+			iblocklengthFinal = byteLenOfPlaintext * 2;
+		
+		
+		
+		String plaintextHex = guiHandler.rabinFirst.bytesToString(plaintext.getBytes());
+		String paddingScheme = cstb.cmbChooseBlockPadding.getItem(cstb.cmbChooseBlockPadding.getSelectionIndex());
+		
+		//String paddedPlaintextHex = guiHandler.rabinFirst.getPaddedPlaintext(plaintextHex, guiHandler.bytesPerBlock, paddingScheme);
+		String paddedPlaintextHex = guiHandler.rabinFirst.getPaddedPlaintext(plaintextHex, maxBytesPerBlock, paddingScheme);
+		ArrayList<String> plaintextsHex = guiHandler.rabinFirst.parseString(paddedPlaintextHex, maxBytesPerBlock);
+		
+		// test plaintext with prefix padding
+		//ArrayList<String> prefixPaddedPlaintextblocks = guiHandler.rabinFirst.getPaddedPlaintextblocks(plaintextsHex, guiHandler.blocklength);
+		ArrayList<String> prefixPaddedPlaintextblocks = guiHandler.rabinFirst.getPaddedPlaintextblocks(plaintextsHex, iblocklengthFinal);
+		paddedPlaintextHex = guiHandler.rabinFirst.getArrayListToString(prefixPaddedPlaintextblocks);
+		cstb.txtPlaintext.setText(paddedPlaintextHex);
+		
+		//ciphertextList = guiHandler.rabinFirst.getCiphertextblocksAsList(plaintext, paddingScheme, guiHandler.bytesPerBlock, guiHandler.blocklength, guiHandler.radix);
+		ciphertextList = guiHandler.rabinFirst.getEncryptedListOfStrings(plaintextsHex, guiHandler.radix);
+		ArrayList<String> paddedCiphertextsHex = guiHandler.rabinFirst.getPaddedCiphertextblocks(ciphertextList, iblocklengthFinal);
+		ciphertextList = paddedCiphertextsHex;
+		String ciphertext = guiHandler.rabinFirst.getArrayListToString(ciphertextList);
+		//String ciphertext = guiHandler.rabinFirst.getArrayListToString(paddedCiphertextsHex);
+		cstb.txtCiphertext.setText(ciphertext);
+		//System.out.println(ciphertextList.size());
+		cstb.btnEncrypt.setEnabled(false);
+		
+		
+	}
+	
+	
+	private void initializeClickedPlaintexts(int size, int sizePlaintexts) {
+		clickedPlaintexts = new ArrayList<ArrayList<Boolean>>();
+		for(int i = 0; i < size; i++) {
+			ArrayList<Boolean> initState = new ArrayList<Boolean>();
+			for(int j = 0; j < sizePlaintexts; j++) {
+				initState.add(false);
+			}
+			clickedPlaintexts.add(initState);
+		}
+	}
+	
+		
+	
+	private void setChooseBlock(Combo cmbChooseBlock, int size) {
+		cmbChooseBlock.removeAll();
+		for(int i = 1; i <= size; i++) {
+			cmbChooseBlock.add(String.valueOf(i));
+		}
+	}
+	
+	
+	private void markCiphertext(StyledText ciphertext, int elem) {
+		int startIdx = guiHandler.getStartIdx(elem, ciphertextList);
+		int endIdx = guiHandler.getEndIdx(startIdx, elem, ciphertextList);
+		ciphertext.setSelection(startIdx, endIdx);
+	}
+	
+	
+	private void setPossiblePlaintexts(int idx, Text txtFirstPlaintext, Text txtSecondPlaintext, 
+			Text txtThirdPlaintext, Text txtFourthPlaintext) {
```

Comment:
Methods with longer/more parameters in Java are best formatted like

```java
// Option 2
void method1(
        JavaStringMethodFactoryBean param1, String param2, String param3
) {

// Option 2
void method1(
        JavaStringMethodFactoryBean param1,
       String param2,
       String param3
) {
```

```suggestion
	private void setPossiblePlaintexts(
	        int idx, Text txtFirstPlaintext, Text txtSecondPlaintext, Text txtThirdPlaintext, Text txtFourthPlaintext
	) {
```